common/static/common/js/discussion/views/discussion_thread_list_view.js

```diff
@@ -493,29 +509,116 @@
                 return crumbs;
             };
 
-            DiscussionThreadListView.prototype.filterTopics = function(event) {
+            DiscussionThreadListView.prototype.selectOption = function($element) {
+                var activeDescendantId, activeDescendantText;
+                if (this.selectedTopic.length > 0) {
+                    this.selectedTopic.removeClass('is-focused');
+                }
+                if ($element) {
+                    $element.addClass('is-focused');
+                    activeDescendantId = $element.parent().attr('id');
+                    activeDescendantText = $element.text();
+                    this.selectedTopic = $element;
+                    this.selectedTopicId = activeDescendantId;
+                    $('.forum-nav-browse-filter-input')
+                        .attr('aria-activedescendant', activeDescendantId)
+                        .val(activeDescendantText);
+                }
+            };
+
+            DiscussionThreadListView.prototype.filterInputReset = function() {
+                this.filterEnabled = true;
+                this.selectedTopicIndex = -1;
+                this.selectedTopicId = null;
+            };
+
+            DiscussionThreadListView.prototype.keyboardBinding = function(event) {
+                var $inputText = $('.forum-nav-browse-filter-input'),
+                    $filteredMenuItems = $('.forum-nav-browse-menu-item:visible'),
+                    filteredMenuItemsLen = $filteredMenuItems.length,
+                    $curOption = $filteredMenuItems.eq(0).find('.forum-nav-browse-title').eq(0),
+                    $activeOption, $prev, $next;
+
+                switch (event.keyCode) {
+                case 13: // enter key
```

Comment:
Looks like `constants` is not exposed on the global `edx.util`, so you can't use the keycodes from there without converting to requirejs (which we will do to this file in the coming sprint). In the meantime you could just create a dictionary in the file and reference the key codes as named constants, like Jareer did [here](https://github.com/edx/edx-platform/pull/13435/files#diff-72f9cd39a0a77ed156fe00d3a2aaad97R9)